tests/test_tiktoken.py

```diff
@@ -201,31 +221,17 @@ def test_tiktoken_vocab(model_name: Optional[str], encoding_name: Optional[str],
     reloaded_wrapped_vocab = reloaded_wrapped_tokenizer.get_vocab()
     assert wrapped_vocab == reloaded_wrapped_vocab
 
-    didnt_match = []
     for key, value in wrapped_vocab.items():
         # Skip checking the extra ids we pad the vocab with
         if key.startswith('<extra_id') and key.endswith('>'):
             continue
 
-        if original_tokenizer.encode(key, allowed_special='all') == [value]:
-            continue
-        else:
-            didnt_match.append(
-                (key, original_tokenizer.encode(key,
-                                                allowed_special='all'), value))
-
-    # Decode is lossy because some bytes are not representable in utf-8
-    # see https://github.com/openai/tiktoken/blob/39f29cecdb6fc38d9a3434e5dd15e4de58cf3c80/tiktoken/core.py#L245-L247
-    # This means that the str: int vocab mapping doesn't work. Would have to look more into how other HF tokenizers handle this.
-    model_or_encoding_name = model_name or encoding_name
-    if model_or_encoding_name is not None:
-        expected_didnt_match = MODEL_OR_ENCODING_NAME_TO_NON_UTF8_TOKENS.get(
-            model_or_encoding_name)
-        assert len(didnt_match) == expected_didnt_match
-    else:
-        raise NotImplementedError(
-            'Add the new tokenizer and how many tokens in the vocab are not utf8 representable.'
-        )
+        expected_decoding = ''.join([
+            bytes_to_unicode()[ord(char)]
+            for char in original_tokenizer.decode_single_token_bytes(
+                value).decode('latin-1')
```

Comment:
What is latin-1?

Reply:
https://en.wikipedia.org/wiki/ISO/IEC_8859-1, as mentioned in the main code, I got this tip from someone else's attempt to create an HF compatible gpt4 tokenizer (https://gist.github.com/xenova/a452a6474428de0182b17605a98631ee), and it seems to match the implementation of existing HF tokenizers.